src/main/java/com/example/gradecalculator/controller/UserController.java

```diff
@@ -76,6 +90,22 @@ public String userGet(Model model) {
         return "user";
     }
 
+    @PostMapping("/editProfile")
+    private String editProfilePost(@Valid @ModelAttribute UserEditTO editProfile, Authentication authentication){
+
+        var userService = (UserDetailsImpl)authentication.getPrincipal();
+        var userResult = userRepository.findById(userService.getId());
```

Comment:
As this is the second time this is used in the same class, make it a dedicated method.<br>
You can also make an HelperClass that holds methods for the principal (= athenticated user)